crates/compiler/can/src/desugar.rs

```diff
@@ -280,6 +279,19 @@ pub fn desugar_value_def_suffixed<'a>(arena: &'a Bump, value_def: ValueDef<'a>)
                         ),
                     },
                 ),
+                // When the last expression is suffixed, it will try to unwrap the def, but because we
+                // have an annotated def we can simply ignore the try and return it as is without
+                // creating an intermediate identifier
+                Err(EUnwrapped::UnwrappedDefExpr { loc_expr, .. }) => desugar_value_def_suffixed(
```

Comment:
I'm not a huge fan of using a syntax error as special state to decide when to successfully continue, but:
1) We probably do this elsewhere
2) This code is going away with purity inference and the new `try` keyword

So it doesn't matter enough to be worth making much cleaner.